src/script/filter.js

```diff
@@ -20,31 +23,51 @@ export class Filter {
     */
 
     filter () {
-        const renderer = new Renderer();
+        const renderer = new Renderer(this.data);
+        const filterByTag = new FilterByTag();
+        const filterByType = new FilterByType();
+        const filerByRating = new FilterByRating();
         const filteredList = { challenges: [],};
 
-        this.data.challenges.forEach(challenge => {
+        // Checks all filter elements if they are empty, if they all are empty (returning false) load all challenges
+        if ( 
+            filterByTag.checkDOM() ||
+            filterByType.checkDOM() ||
+            filerByRating.checkDOM()
+            ){
+            
 
-            /* if( filterByTag(challenge) &&
-                filterByText(challenge) && 
-                filterByType(challenge) && 
-                filterByType(challange) )  */
-            if (filterByTag(challenge)) {
 
-                filteredList.challenges.push(challenge);    
-            }
-        })
+            this.data.challenges.forEach(challenge => {
+                // Checks for each filter if the challenge matches the filter (returns true) or if all the DOM elements for that filter is empty
+                if (
+                    (filterByTag.filter(challenge) || !filterByTag.checkDOM()) &&
```

Comment:
I can't see a reason for the two separate methods ( `filter()` and `checkDOM()` respectively). As far as I can tell, they are always invoked in tandem, like here, and they use the same data and have roughly the same signature (returning a boolean). So as far as I can tell, they could have just been a single method.

Furthermore, both names are confusing to me. The word `filter()` implies a list, but takes a single challenge, and I don't understand from looking here what `checkDOM()` is for, but looking at the code for it in some of the filters confuse me even more because some of them don't even look at the DOM.